libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -0,0 +1,162 @@
+        mock_client.get_blob_client.side_effect = mock_blob_clients
+        mock_container_client_cls.return_value = mock_client
+        yield mock_container_client_cls, mock_client
+
+
+def test_lazy_load(
+    account_url: str,
+    container_name: str,
+    blobs: list[dict[str, str]],
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    loader = create_azure_blob_storage_loader()
+    docs = loader.lazy_load()
+    assert len(list(docs)) == len(blobs)
+    for index, doc in enumerate(docs):
+        assert doc.page_content == blobs[index]["blob_content"]
+        assert (
+            doc.metadata["source"]
+            == f"{account_url}/{container_name}/{blobs[index]['blob_name']}"
+        )
+
+
+def test_get_blob_client(
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+    mock_container_client: Tuple[MagicMock, MagicMock],
+) -> None:
+    _, mock_client = mock_container_client
+    mock_client.list_blob_names.return_value = ["text_file.txt"]
+
+    loader = create_azure_blob_storage_loader(prefix="text")
+    list(loader.lazy_load())
+    mock_client.get_blob_client.assert_called_once_with("text_file.txt")
+
+
+def test_default_credential(
+    mock_container_client: Tuple[MagicMock, MagicMock],
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    mock_container_client_cls, _ = mock_container_client
+    loader = create_azure_blob_storage_loader(blob_names="text_file.txt")
+    list(loader.lazy_load())
+    cred = mock_container_client_cls.call_args[1]["credential"]
+    assert isinstance(cred, azure.identity.DefaultAzureCredential)
+
+
+def test_override_credential(
+    mock_container_client: Tuple[MagicMock, MagicMock],
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    from azure.core.credentials import AzureSasCredential
+
+    mock_container_client_cls, _ = mock_container_client
+    mock_credential = AzureSasCredential("test_sas_token")
+    loader = create_azure_blob_storage_loader(
+        blob_names="text_file.txt", credential=mock_credential
+    )
+    list(loader.lazy_load())
+    assert mock_container_client_cls.call_args[1]["credential"] is mock_credential
+
+
+def test_async_credential_provided_to_sync(
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    from azure.core.credentials_async import AsyncTokenCredential
+
+    mock_credential = MagicMock(spec=AsyncTokenCredential)
+    loader = create_azure_blob_storage_loader(
+        blob_names="text_file.txt", credential=mock_credential
+    )
+    with pytest.raises(ValueError, match="Cannot use synchronous load"):
+        list(loader.lazy_load())
+
+
+def test_invalid_credential_type(
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    mock_credential = MagicMock(spec=str)
```

Comment:
Instead of wrapping this with a mock, let's just use a string (e.g., `"<account-key>") as we don't really need any mock functionality here